ecomap/ecomap/AddProblem/Controllers/AddProblemViewController.h

```diff
@@ -0,0 +1,14 @@
+//
+//  AddProblemViewController.h
+//  ecomap
+//
+//  Created by Anton Kovernik on 10.02.15.
+//  Copyright (c) 2015 SoftServe. All rights reserved.
+//
+
+#import "MapViewController.h"
+#import "Defines.h"
+
+@interface AddProblemViewController : MapViewController <UIPickerViewDataSource, UIPickerViewDelegate, UIImagePickerControllerDelegate, UINavigationControllerDelegate, UIScrollViewDelegate>
```

Comment:
Have no how it works. How it still works.... Maybe because there are no so many functionality, but it definitely will be broken with functionality increasing or with some refactoring. First smell is in conforming too many protocols. You should create separate view controller for every view and after put them into scroll view or something like this. Currently this view controller have to many logic inside.